magma/circuit.py

```diff
@@ -596,8 +596,12 @@ def DefineCircuit(name, *decl, **args):
     currentDefinition = DefineCircuitKind( name, (Circuit,), dct)
     return currentDefinition
 
+__seenCircuitNames = {}
 def EndDefine():
     if currentDefinition:
+        if currentDefinition.name in __seenCircuitNames:
+            raise Exception(f"The circuit name '{currentDefinition.name}' was already used. Module names must be unique!")
+        __seenCircuitNames[currentDefinition.name] = 1
```

Comment:
For now, could we add this as a configurable option? That way we can merge it into the mainline release ASAP but then figure out how exactly we want to handle it and what the default behavior should be (I expect there to be some discussion).

We could add a simple API like `set`/`get_debug_mode`  https://github.com/phanrahan/magma/blob/master/magma/config.py#L14-L24 to enable this?

Reply:
IMO I would rather have the discussion and decide to fix the issue or not. We don't need to have this merged in immediately. What do people think? Are there any arguments on the other side?